pkg/operator/ceph/pool/dependents.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	v1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/util/dependents"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CephBlockPoolDependents returns the rbd namespaces (s) which exist in the rbd pool that should block
+// deletion.
+func CephBlockPoolDependents(clusterdCtx *clusterd.Context, clusterInfo *client.ClusterInfo, blockpool *v1.CephBlockPool) (*dependents.DependentList, error) {
+	nsName := fmt.Sprintf("%s/%s", blockpool.Namespace, blockpool.Name)
+	baseErrMsg := fmt.Sprintf("failed to get dependents of CephBlockPool %q", nsName)
+
+	deps := dependents.NewDependentList()
+
+	// CepbBlockPoolNamespaces
+	namespaces, err := clusterdCtx.RookClientset.CephV1().CephBlockPoolRadosNamespaces(blockpool.Namespace).List(clusterInfo.Context, metav1.ListOptions{})
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to list CephBlockPoolRadosNamespaces for CephBlockPool %q", baseErrMsg, nsName)
+	}
+	for _, namespace := range namespaces.Items {
```

Comment:
we do expect many namespaces per pool. with this option, a user can create a single pool and create `N` number of rados namespaces to provide isolation between the users.

Reply:
>Instead of looping through things, we could add a list of rados namespaces to respective CephBlockPool status? While deleting we just ensure that the list is empty. We manipulate the list from rados namespace controller. This also makes monitoring easier.

the list of namespaces doesn't have any limits on numbers and also this increases the contents of the block CR and managing it might become complicated.

>Also, since rados namespaces are parts of block pool, adding ownerRefs helps achieve intended deletion protection.

AFAIK This is the dependency logic followed in the Rook https://github.com/rook/rook/blob/476e94632bdd41e4aca64dcd51f4ac6b1e7f4eb7/pkg/operator/ceph/file/dependent.go#L32-L51